net/tcp/tcp_close.c

```diff
@@ -38,69 +38,41 @@
  ****************************************************************************/
 
 #include <nuttx/config.h>
+#ifdef CONFIG_NET_TCP
 
-#include <sys/types.h>
-#include <sys/socket.h>
-#include <stdint.h>
-#include <stdbool.h>
 #include <errno.h>
 #include <debug.h>
 #include <assert.h>
 
-#include <arch/irq.h>
-
-#include <nuttx/semaphore.h>
 #include <nuttx/net/net.h>
 #include <nuttx/net/netdev.h>
 #include <nuttx/net/tcp.h>
-#include <nuttx/net/udp.h>
 
 #include "netdev/netdev.h"
 #include "devif/devif.h"
 #include "tcp/tcp.h"
-#include "udp/udp.h"
-#include "pkt/pkt.h"
-#include "local/local.h"
 #include "socket/socket.h"
-#include "usrsock/usrsock.h"
-#include "inet/inet.h"
 
 /****************************************************************************
  * Private Types
  ****************************************************************************/
 
-#ifdef NET_TCP_HAVE_STACK
 struct tcp_close_s
 {
   FAR struct devif_callback_s *cl_cb;     /* Reference to TCP callback instance */
   FAR struct socket           *cl_psock;  /* Reference to the TCP socket */
   sem_t                        cl_sem;    /* Signals disconnect completion */
   int                          cl_result; /* The result of the close */
 };
-#endif
```

Comment:
Why do you remove these dependencies on NET_TCP_HAVE_STACK.  Doesn't that break the USRSOCK case.  That is what those defines are for?

Reply:
Did you move these checkes elsewhere?

Reply:
inet_close.c doesn't exist anymore, the code move into tcp/tcp_close.c or udp/udp_close.c. Since tcp/udp folder don't compiled if only USRSOCK is enabled, that is why I remove the check.

Reply:
Since inet_close.c is very similar to tcp_close.c, git consider it is the move operation and show the difference between inet_close.c(removed) and tcp_close.c(new), but tcp_close.c don't compile at all if CONFIG_NET_TCP isn't enabled.